Source/WebCore/dom/Element.h

```diff
@@ -197,7 +197,7 @@ class Element : public ContainerNode {
 
     void scrollBy(const ScrollToOptions&);
     void scrollBy(double x, double y);
-    virtual void scrollTo(const ScrollToOptions&, ScrollClamping = ScrollClamping::Clamped, ScrollSnapPointSelectionMethod = ScrollSnapPointSelectionMethod::Closest);
+    virtual void scrollTo(const ScrollToOptions&, ScrollClamping = ScrollClamping::Clamped, ScrollSnapPointSelectionMethod = ScrollSnapPointSelectionMethod::Closest, ScrollClass scrollClass = ScrollClass::ScrollTo, FloatPoint originalScrollDelta = { });
```

Comment:
Scroll delta should be FloatSize.

Reply:
Also originalScrollDelta is only non-zero when ScrollClass is ScrollBy, so there's redundancy here. You could use std::optional<FloatSize> to indicate it's a delta, and remove ScrollClass?

Reply:
Fixed to use a std::optional<FloatSize>.